packages/mobile/src/account/Profile.tsx

```
@@ -1,64 +1,105 @@
+import { useDispatch, useSelector } from 'react-redux'
+import { setName, setPicture } from 'src/account/actions'
+import { nameSelector, pictureSelector } from 'src/account/selectors'
+import { showMessage } from 'src/alert/actions'
+import CancelButton from 'src/components/CancelButton'
+import i18n, { Namespaces } from 'src/i18n'
+import { emptyHeader } from 'src/navigator/Headers'
+import { Screens } from 'src/navigator/Screens'
+import { TopBarTextButton } from 'src/navigator/TopBarButton'
+import { StackParamList } from 'src/navigator/types'
+import PictureInput from 'src/onboarding/registration/PictureInput'
+import { saveProfilePicture } from 'src/utils/image'
 
-interface StateProps {
-  name: string | null
-  userContact: UserContactDetails
-}
+type Props = StackScreenProps<StackParamList, Screens.Profile>
 
-interface DispatchProps {
-  setName: typeof setName
-}
+function Profile({ navigation, route }: Props) {
+  const { t } = useTranslation(Namespaces.accountScreen10)
+  const [newName, setNewName] = useState(useSelector(nameSelector) ?? '')
+  const picturePath = useSelector(pictureSelector)
+  const [newPictureUri, setNewPictureUri] = useState(picturePath)
 
-// tslint:disable-next-line: no-empty-interface
-interface OwnProps {}
+  const dispatch = useDispatch()
 
-type Props = OwnProps & StateProps & DispatchProps & WithTranslation
-const mapStateToProps = (state: RootState) => {
-  return {
-    name: state.account.name,
-    userContact: userContactDetailsSelector(state),
+  useEffect(() => {
+    if (route.params?.save) {
+      navigation.setParams({ save: false })
+      dispatch(setName(newName))
+      dispatch(setPicture(newPictureUri))
+      // TODO: Save name and picture on CIP-8.
+      dispatch(showMessage(t('namePictureSaved')))
+      navigation.goBack()
+
+      // Delete old proflie pictures if necessary.
+      if (picturePath && picturePath !== newPictureUri) {
+        RNFS.unlink(picturePath).catch()
```

We should log something if it fails. Swallowed errors make debugging harder.

---

I didn't log anything because this felt more of a nice to have than something necessary, but yeah, there's no cost to having a log anyways.